sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzMonNewListExtensions.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.CompilerServices;
+
+namespace Azure.Monitor.OpenTelemetry.Exporter.Internals;
+
+internal static class AzMonNewListExtensions
+{
+    ///<summary>
+    /// Gets http request url from activity tag objects.
+    ///</summary>
+    internal static string? GetNewSchemaRequestUrl(this AzMonList tagObjects)
+    {
+        try
+        {
+            var serverAddress = AzMonList.GetTagValue(ref tagObjects, SemanticConventions.AttributeServerAddress)?.ToString();
+            if (serverAddress != null)
+            {
+                UriBuilder uriBuilder = new()
```

Comment:
Not for this PR: Would StringBuilder be more efficient than this?

Reply:
Some of these Uri classes perform extra validation to confirm that we're building a valid url.

Reply:
Uri builder is better here. We have to handle multiple scenario here. The only change that is required in follow up PR is to use GetTagValues instead of GetTagValue.

Reply:
If the attribute values are as per spec then it should be valid right? Do we need those extra validations if it is not validated by ingestion?

Reply:
e.g. from instrumentation
https://github.com/open-telemetry/opentelemetry-dotnet/blob/ad08228dfec8fd7d8839e09f68e9f0caef4233f4/src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs#L426-L470

Reply:
All of it are not required field, hence we need to check for null or empty on them.

Reply:
seems much faster
![image](https://github.com/Azure/azure-sdk-for-net/assets/17655898/afe7181b-a9d7-4b2d-876f-fc63eb2c83d1)

Reply:
Will include in follow up PR.